### PR TITLE
Trie

### PR DESCRIPTION
**Pull request by morganelle**
- python implementation of a trie data structure
- tests and README included

### CODE REVIEW

File: src/test_trie.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+@pytest.fixture
+def init_jumbo_list():
+    """Fixture for 250k word test."""
+    # jumbo_trie = Trie()
+    word_file = open('/usr/share/dict/words', 'r')
+    word_text = word_file.read()
+    word_file.close()
+    word_text = word_text.split('\n')
+    # for word in word_text:
```
Comment (clair3st) [inline src/test_trie.py:38]: Delete commented out code

File: src/test_trie.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    new_trie.insert('bee')
+    new_trie.insert('bell')
+    return new_trie
+
+
+@pytest.fixture
+def init_jumbo_list():
+    """Fixture for 250k word test."""
+    # jumbo_trie = Trie()
+    word_file = open('/usr/share/dict/words', 'r')
```
Comment (clair3st) [inline src/test_trie.py:34]: There is a cleaner way to do these few lines, take a look into the `with open(<filename>, 'r'):` command. Using this you don't have to explicitly close the file.

File: src/test_trie.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+def test_insert_two_words(init_empty_trie):
+    """Check error is raised when iterable inserted."""
+    with pytest.raises(ValueError):
+        init_empty_trie.insert('hi there')
+
+
+def test_insert_duplicate(init_trie_one_word):
+    """Check error is raised when iterable inserted."""
+    with pytest.raises(ValueError):
```
Comment (clair3st) [inline src/test_trie.py:87]: I don't think you should explicitly raise an error. I think in this situation it should just do nothing, That way it won't stop a program from running if you insert the same word.

File: src/test_trie.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+def test_size_trie_before_and_after_insert(init_trie_three_words_shared_letters):
+    """Test trie size before and after a removal."""
+    assert init_trie_three_words_shared_letters.size() == 3
+    init_trie_three_words_shared_letters.insert('happy')
+    assert init_trie_three_words_shared_letters.size() == 4
+
+
+def test_insert_jumbo_list(init_jumbo_list):
+    """."""
```
Comment (clair3st) [inline src/test_trie.py:191]: docstring

File: src/test_trie.py [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    jumbo_trie = Trie()
+    for word in init_jumbo_list:
+        jumbo_trie.insert(word)
+    assert len(init_jumbo_list) == jumbo_trie.size()
+    assert jumbo_trie.contains('apple')
```
Comment (clair3st) [inline src/test_trie.py:196]: I would also write a test for remove that uses the large trie. Also test that if you remove a larger word like 'apples' the word 'apple' still exists.

File: src/test_trie.py (hunk 2 above)
Comment (morganelle) [reply src/test_trie.py:34]: Updated in PR #18

File: src/test_trie.py (hunk 1 above)
Comment (morganelle) [reply src/test_trie.py:38]: Updated in PR #18

File: src/test_trie.py (hunk 4 above)
Comment (morganelle) [reply src/test_trie.py:191]: Updated in PR #18

File: src/test_trie.py (hunk 5 above)
Comment (morganelle) [reply src/test_trie.py:196]: Updated in PR #18

**Pull request closed without merging**